cupyx/scipy/ndimage/measurements.py

```diff
@@ -200,3 +203,297 @@ def _kernel_finalize():
         y[i] = j + 1;
         ''',
         'cupyx_nd_label_finalize')
+
+
+_ndimage_variance_kernel = cupy.ElementwiseKernel(
+    'T input, R labels, raw X index, int32 size, raw Z mean',
+    'raw Z out',
+    """
+    for (ptrdiff_t j = 0; j < size; j++) {
+      if (labels == index[j]) {
+        atomicAdd(&out[j], (input - mean[j]) * (input - mean[j]));
+        break;
+      }
+    }
+    """)
+
+
+_ndimage_sum_kernel = cupy.ElementwiseKernel(
+    'T input, R labels, raw X index, int32 size',
+    'raw Y out',
+    """
+    for (ptrdiff_t j = 0; j < size; j++) {
+      if (labels == index[j]) {
+        atomicAdd(&out[j], input);
+        break;
+      }
+    }
+    """)
+
+
+def _ndimage_sum_kernel_2(input, labels, index, out, batch_size=4):
+    for i in range(0, index.size, batch_size):
+        matched = labels == index[i:i + batch_size].reshape(
+            (-1,) + (1,) * input.ndim)
+        sum_axes = tuple(range(1, 1 + input.ndim))
+        out[i:i + batch_size] = cupy.where(matched, input, 0).sum(
+            axis=sum_axes)
+    return out
+
+
+_ndimage_mean_kernel = cupy.ElementwiseKernel(
+    'T input, R labels, raw X index, int32 size',
+    'raw Y out, raw int32 count',
```

Comment:
`int32` may be too small when, for example, computing an input with its shape `(4096, 4096, 4096)`.
```suggestion
    'raw Y out, raw uint64 count',
```